### PR TITLE
Fix bytes/unicode issues for Python 3 in tryclient

### PR DESCRIPTION
**Pull request by rodrigc**
## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation
---
Fixes https://github.com/buildbot/buildbot/issues/4037

### CODE REVIEW

**Comment by tardyp** [comment]
probably we should add a minimal integration test that runs a try command

**Comment by tardyp** [comment]
@rodrigc you left IRC. I am not sure where you need help to finish this.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # All files to globally ignore
+.vscode
```
Comment (sa2ajj) [inline .gitignore]: should we add this at the end?..

File: master/buildbot/clients/tryclient.py [hunk 2]
```diff
@@ -51,7 +52,7 @@ def __init__(self, branch, revision, patch, repository=''):
 
 
 def output(*msg):
-    log.msg(' '.join(map(str, msg)))
+    print(' '.join([str(m)for m in msg]))
```
Comment (sa2ajj) [inline master/buildbot/clients/tryclient.py]: this looks like a separate commit to me...

File: master/buildbot/clients/tryclient.py [hunk 3]
```diff
@@ -349,23 +350,23 @@ def readConfig(self):
 
     def parseConfig(self, res):
         self.config = {}
-        for l in res.split("\n"):
+        for l in res.split(b"\n"):
```
Comment (sa2ajj) [inline master/buildbot/clients/tryclient.py]: could we use a bit more descriptive `line` here?

File: master/buildbot/clients/tryclient.py [hunk 4]
```diff
@@ -349,23 +350,23 @@ def readConfig(self):
...
                 if len(parts) < 2:
                     parts.append('true')
                 self.config[parts[0]] = parts[1]
         return self.config
 
     def parseTrackingBranch(self, res):
         # If we're tracking a remote, consider that the base.
-        remote = self.config.get("branch." + self.branch + ".remote")
-        ref = self.config.get("branch." + self.branch + ".merge")
+        remote = self.config.get(b"branch." + self.branch + b".remote")
```
Comment (sa2ajj) [inline master/buildbot/clients/tryclient.py]: the expression is repeated here, a single variable would be better to avoid c&p errors

Comment (sa2ajj) [reply master/buildbot/clients/tryclient.py]: oops, i misread the code :(

**Comment by sa2ajj** [comment]
And it looks like the branch needs a rebase...

File: .gitignore (hunk 1 above)
Comment (rodrigc) [reply .gitignore]: Removed this

**Pull request merged**